test/test_kubeclient.rb

```diff
@@ -389,16 +400,18 @@ def test_api_bearer_token_success
   def test_api_bearer_token_failure
     error_message = '"/api/v1" is forbidden because ' \
                     'system:anonymous cannot list on pods in'
-    response = OpenStruct.new(code: 401, message: error_message)
+    response = RestClientResponseStub.new('', 403)
```

Comment:
The previous code was raising a `KubeException` in the request, which meant that `#handle_exception` was just passing through the exception. So all the expectations in this test were just testing the `OpenStruct`, not testing the code within `#handle_exception` that retains the status code and response body.

Instead of raising a `KubeException` this raises the `RestClient::Exception` which is what `#handle_exception` is rescuing and parsing. The response is required by `RestClient::Exception`. The stub extends `String` because the `RestClient::Response` object extends `String` and is expected to behave the same way (that is, the response _is_ the body).